### DISCUSSION TITLE
Feature Request: Add functionality in the portfile to allow app-local dll copy that is dynamic loaded

### DISCUSSION THREAD

**Discussion by jasjuang** [microsoft/vcpkg#17770] [Ideas]
I am still trying to make a portfile for [RedSDK](https://www.red.com/developers), and as mentioned in #2313 part of the SDK is closed source. Meaning they only provide some cpp files and some lib files and dll files. One new problem I encountered is the dependent dlls is actually dynamically loaded instead of dynamic linked, which means when I link RedSDK on my downstream project, the dependent dlls would not be copied to my local folder, which is expected because the app-local dll copy mechanism from vcpkg is using dumpbin to investigate the dependencies, and the dependent dlls would not show up because it is dynamically loaded instead of dynamic linked.

My understanding is there isn't a way to automatically discover the dll dependencies if it's dynamic loaded. If my understanding is correct, can we add a functionality so we can manually specify something in the portfile so vcpkg will perform the app-local copy?

Slightly related to #2343

**Comment by ras0219-msft** [comment]
Thanks for bringing this up!

For some libraries, I think this can be done by applying a similar technique as https://github.com/Microsoft/vcpkg/blob/d88563cd095b9aaad81d57f1c0a254d7e17cf859/ports/qt5/qtdeploy.ps1 (though we need to make it more "expandable" for arbitrary ports to contribute). This should work as long as the user's code is linking a "known" dll that we can key off of.

Would that work for you?

**Comment by jasjuang** [comment]
Yes if we can create a hard-coded deploy patch for each individual port that will work for now that will still be great. 

However I am not really familiar with powershell scripting language so it might take me some time to figure it out.

**Comment by NancyLi1013** [comment]
This issue hasn’t been updated in a year; if it is still an issue, please reopen.

**Comment by jasjuang** [comment]
It's still an issue. Currently, I still have to manually copy the dlls that are dynamically loaded. Still looking forward to having an automated solution.
